api/pkg/transformer/pipeline/util.go

```diff
@@ -1,20 +1,24 @@
 package pipeline
 
 import (
-	"fmt"
-
 	"github.com/antonmedv/expr"
+
+	mErrors "github.com/gojek/merlin/pkg/errors"
 	"github.com/gojek/merlin/pkg/transformer/types/operation"
 	"github.com/gojek/merlin/pkg/transformer/types/series"
 )
 
 func evalJSONPath(env *Environment, jsonPath string) (interface{}, error) {
 	c := env.CompiledJSONPath(jsonPath)
 	if c == nil {
-		return nil, fmt.Errorf("compiled jsonpath %s not found", jsonPath)
+		return nil, mErrors.NewInvalidInputErrorf("compiled jsonpath %s not found", jsonPath)
```

Comment:
I think it's not invalid input error since the compilation of json path is completely handled by us.

Reply:
Suggestion: Before running env.CompiledJSONPath(), we can validate if `jsonPath` is not an empty string.

```
if jsonPath == "" {
        return nil, mErrors.NewInvalidInputErrorf("jsonpath is not specified")
}

c := env.CompiledJSONPath(jsonPath)
if c == nil {
        return nil, fmt.Errorf("compiled jsonpath %s not found", jsonPath)
}
```